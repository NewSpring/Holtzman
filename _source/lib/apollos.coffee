###

  Apollos.name

  @example get a string with the name of this system

    console.log Apollos.name

###
Apollos.name = "Apollos"


###

  Apollos.user

  @example get the currently logged in user

    console.log Apollos.user()._id

###
Apollos.user = (id) ->

  if Meteor.isServer
    return {}

  if Meteor.isClient
    user = Meteor.user()

  return user or {}


Apollos.user.login = {}

if Meteor.isClient
  Meteor.subscribe "currentUserPerson"
else
  Meteor.publish "currentUserPerson", ->
    user = Apollos.users.findOne @.userId
    query = {}
    queryOk = false

    if not user
      return

    if user.personGuid
      query.guid = user.personGuid
      queryOk = true

    # we need to move the rock stuff out of this package
    if user.rock and user.rock.personId
      query.personId = user.rock.personId
      queryOk = true

    if not queryOk
      return

    return Apollos.people.find query


###

  Apollos.person

  @example get the currently logged in user's person document

    console.log "Hello, #{Apollos.person().firstName}"

###
Apollos.person = (user) ->

  if Meteor.isServer
    if not user?._id
      throw new Meteor.Error ("user is requried for server lookup")
      return

  if Meteor.isClient
    user or= Apollos.user()

  if user.personGuid
    person = Apollos.people.findOne
      guid: user.personGuid


  # we shouldnt have any code for rock in the core package
  # this will probably need to adjust the rock package

  # if not person and user.rock and user.rock.personId
  #   person = Apollos.people.findOne
  #     personId: user.rock.personId

  return person or {}






if Meteor.server
  Accounts.onCreateUser (options, user) ->

    if options.profile
      user.profile = options.profile

    if user.profile?.guest is true
      return user

    person = Apollos.person user

    # # no existing user so create one
    if not Object.keys(person).length
      if not user.personGuid
        user.personGuid = Apollos.utilities.makeNewGuid()

      Apollos.people.upsert({guid: user.personGuid}, {
        $set:
          preferredEmail: user.emails[0].address
      })

    return user

###

  Apollos.user.create

  @example create a new user

    Apollos.user.create "xyz@abc.cc", "password1", (error) ->
      if error
        console.log error

  @param email is the email address for the user to use to login with
  @param password is the plain-text password the user will authenticate with
  @param callback is the function that will be called with an error if so

###
Apollos.user.create = (email, password, callback) ->

  Accounts.createUser
    email: email
    password: password
  ,
    callback

###

  Apollos.user.forgotPassword

  @example request a forgot password email

    Apollos.user.forgotPassword "xyz@abc.cc", (error) ->
      if error
        console.log error

  @param email is the email address for the user to use to login with
  @param callback is the function that will be called with an error if so

###
Apollos.user.forgotPassword = (email, callback) ->

  return Accounts.forgotPassword
    email: email
  ,
    callback

###

  Apollos.user.resetPassword

  @example changes password using reset token

    Apollos.user.resetPassword token, newPassword, (error) ->
      if error
        console.log error

  @param token is the reset token emailed to the user
  @param newPassword is what to change the password to
  @param callback is the function that will be called with an error if so

###
Apollos.user.resetPassword = (token, newPassword, callback) ->

  return Accounts.resetPassword token, newPassword, callback


###

  Apollos.user.changePassword

  @example changes password using old password

    Apollos.user.changePassword oldPassword, newPassword, (error) ->
      if error
        console.log error

  @param oldPassword is the user's current password
  @param newPassword is what to change the password to
  @param callback is the function that will be called with an error if so

###
Apollos.user.changePassword = (oldPassword, newPassword, callback) ->

  return Accounts.changePassword oldPassword, newPassword, callback

if Meteor.isServer
  # TODO this is not secure
  Apollos.emailTemplates = Accounts.emailTemplates
