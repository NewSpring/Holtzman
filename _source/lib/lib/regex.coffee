_addRegexTestFunction = (regex) ->
  funcName = "is#{regex.charAt(0).toUpperCase()}#{regex.slice(1)}"
  Apollos.validate[funcName] = (str) ->
    return Apollos.regex[regex].test str


Apollos.regex =

  # @TODO choose which email regex to use
  #
  # this was ours
  # email: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  #
  # this is what rock uses:
  email: /[\w\.\'_%-]+(\+[\w-]*)?@([\w-]+\.)+[\w-]+/

  bcrypt: /^\$2a\$10\$[\/\.a-zA-Z0-9]{53}$/

  phoneNumber: /^[1-9]([0-9]{6}|[0-9]{9})$/

  guid: /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i

  visa: /^4[0-9]{12}(?:[0-9]{3})?$/

  mastercard: /^5[1-5][0-9]{14}$/

  americanExpress: /^3[47][0-9]{13}$/

  discover: /^6(?:011|5[0-9]{2})[0-9]{12}$/

  startOfVisa: /^4[0-9]{0,15}$/

  startOfMastercard: /^5$|^5[1-5][0-9]{0,14}$/

  startOfAmEx: /^3$|^3[47][0-9]{0,13}$/

  startOfDiscover: /^6$|^6[05]$|^601[1]?$|^65[0-9][0-9]?$|^6(?:011|5[0-9]{2})[0-9]{0,12}$/


Apollos.addRegex = (obj) ->

  for key, value of obj
    Apollos.regex[key] = value

  return


Apollos.validate =

  acceptedCardTypes: ["Visa", "Mastercard", "AmEx", "Discover"]

  guessCardType: (number) ->

    number = number.replace /-/g, ''

    for cardType in Apollos.validate.acceptedCardTypes

      if Apollos.validate["isStartOf#{cardType}"] number
        return cardType

    return false


  getCardType: (number) ->

    number = number.replace /-/g, ''

    for cardType in Apollos.validate.acceptedCardTypes

      if Apollos.validate["is#{cardType}"] number
        return cardType

    return false


  isBcryptHash: (str) ->
    return @.isBcrypt(str)


for regex of Apollos.regex
  _addRegexTestFunction regex
