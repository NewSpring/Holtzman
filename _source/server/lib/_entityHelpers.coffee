Apollos.documentHelpers =


  translate: (singular, data, platform) ->

    if not platform
      Apollos.debug "must specify platform to translate to"
      return

    platform = platform.toUpperCase()

    if not Apollos[singular]._dictionary[platform]
      Apollos.debug "no translation found for #{platform} in #{singular}"
      return

    # translate
    return Apollos[singular]._dictionary[platform] data


  ###

    Apollos.documentHelpers.update

    @example update a doc in apollos with data from a platform

      Apollos.documentHelpers.update "person", "people", person, platform

    @param singular [String] The singular form of the entities name
    @param plural [String] The plural form of the entities name
    @param doc [Object] existing doc from other service to be updated
    @param platform [String] platform to be updated from

  ###
  update: (singular, plural, doc, platform) ->

    doc = Apollos[singular].translate doc, platform

    if not doc
      return

    singularIdKeyValue = {}
    singularIdKeyValue["#{singular}Id"] = doc["#{singular}Id"]

    if platform
      doc.updatedBy = platform.toUpperCase()
    else
      doc.updatedBy = Apollos.name

    query =
      $or: [
        singularIdKeyValue
      ,
        guid: RegExp(doc.guid, "i")
      ]

    matches = Apollos[plural].find query,
      sort:
        updatedDate: 1

    if matches.count() > 1
      # Delete older documents, which are the first ones since they are sorted
      ids = matches.map (m) ->
        return m._id

      ids.pop()

      Apollos[plural].remove
        _id:
          $in: ids

      matches = Apollos[plural].find(query)

    if matches.count() is 1
      existing = matches.fetch()[0]
      mongoId = existing._id

      Apollos[plural].update
        _id: mongoId
      ,
        $set: doc

    else
      mongoId = Apollos[plural].insert doc

    return mongoId


  ###

    Apollos.documentHelpers.delete

    @example take an colection and delete it

      Apollos.documentHelpers.delete("person", "people", 3, plaform)

    @param singular [String] The singular form of the entities name
    @param plural [String] The plural form of the entities name
    @param identifier [Object|String|Number] existing document, _id
    @param platform [String] platform initiating the delete

  ###
  delete: (singular, plural, identifier, platform) ->

    if typeof identifier is "number"
      singularIdKeyValue = {}
      singularIdKeyValue["#{singular}Id"] = identifier
      doc = Apollos[plural].findOne singularIdKeyValue

    else if typeof identifier is "string"
      doc = Apollos[plural].findOne identifier

    else
      doc = identifier

    if typeof doc isnt "object"
      throw new Meteor.Error "Delete Error", "Could not delete #{singular}
        identified by #{identifier}"

    if platform
      doc.updatedBy = platform.toUpperCase()
    else
      doc.updatedBy = Apollos.name

    # We have to update this first so the collection hooks know what to do
    Apollos[plural].update
      _id: doc._id
    ,
      $set:
        "updatedBy": doc.updatedBy

    Apollos[plural].remove doc._id
