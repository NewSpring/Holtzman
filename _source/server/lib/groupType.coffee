Apollos.groupType = {}


###

  Apollos.groupType.translate

  @example take data from a service and format it for Apollos

    Apollos.groupType.translate([obj, platform])

  @param groupType [Object] existing object to be translated
  @param platform [String] platform to be translated to

###
Apollos.groupType.translate = (groupType, platform) ->
  return Apollos.documentHelpers.translate "groupType", groupType, platform


###

  Apollos.groupType.update

  @example update a groupType in apollos with data from another service

    Apollos.groupType.update([obj, platform])

  @param groupType [Object] existing groupType from other service to be updated
  @param platform [String] platform to be update from

###
Apollos.groupType.update = (groupType, platform) ->
  return Apollos.documentHelpers.update "groupType", "groupTypes", groupType, platform


###

  Apollos.groupType.delete

  @example take a groupType and delete it

    Apollos.groupType.delete(user, [platform])

  @param groupType [Object|String|Number] existing document, _id
  @param platform [String] platform initiating the delete

###
Apollos.groupType.delete = (groupType, platform) ->
  Apollos.documentHelpers.delete "groupType", "groupTypes", groupType, platform
