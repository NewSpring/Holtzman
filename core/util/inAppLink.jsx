
const inAppLink = (e) => {
  if (Meteor.isCordova) {
    if (cordova.InAppBrowser && cordova.InAppBrowser.open) {
      e.preventDefault()
      cordova.InAppBrowser.open(e.currentTarget.href, "_blank")
    }
  }
}

export default inAppLink
