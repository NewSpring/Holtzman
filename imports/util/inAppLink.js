function openUrl(url, opened, loaded, closed) {
  try {
    SafariViewController.isAvailable(function (available) {
      if (available) {
        SafariViewController.show({
              url: url,
              hidden: false, // default false. You can use this to load cookies etc in the background (see issue #1 for details).
              animated: true, // default true, note that 'hide' will reuse this preference (the 'Done' button will always animate though)
              transition: "curl", // unless animated is false you can choose from: curl, flip, fade, slide (default)
              enterReaderModeIfAvailable: false, // default false
              tintColor: "#6BAC43" // default to ios blue
            },
            // this success handler will be invoked for the lifecycle events 'opened', 'loaded' and 'closed'
            function(result) {
              if (result.event === "opened" && opened) {
                // view has opened
                opened();
              } else if (result.event === "loaded" && loaded) {
                // view has loaded
                loaded();
              } else if (result.event === "closed" && closed) {
                // view has closed
                closed();
              }
            },
            // error function?
            function(msg) {
              alert("KO: " + msg);
            });
      } else {
        // potentially powered by InAppBrowser because that (currently) clobbers window.open
        window.open(url, "_blank", "location=yes");
      }
    });
  }
  catch(err) {
    window.open(url, "_blank", "location=yes");
  }
}

const inAppLink = (e) => {
  if (Meteor.isCordova) {
    if (SafariViewController || (cordova.InAppBrowser && cordova.InAppBrowser.open)) {
      e.preventDefault();
      e.stopPropagation();
      openUrl(e.currentTarget.href);
    }
  }
};

export default inAppLink;

export {
  openUrl,
};
