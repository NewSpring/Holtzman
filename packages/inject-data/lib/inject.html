<script type="text/inject-data"><%= data %></script>
