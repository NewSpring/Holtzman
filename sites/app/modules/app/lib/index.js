

// named collections in EE :(
import Collections from "./collections"

export default {
  Collections
}
