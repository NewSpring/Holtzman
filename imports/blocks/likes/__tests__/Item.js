import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { reset, startBuffering } from "aphrodite/lib/inject";
import mockDate from "mockdate";
jest.mock("../../../util/inAppLink", () => jest.fn());
import inAppLink from "../../../util/inAppLink";
import LikesItem from "../Item";

// XXX god bless you
mockDate.set("1/1/2000");

const defaultProps = {
  like: {
    link: "http://test.com/1",
    category: "test category",
    image: "http://test.com/1.jpg",
    icon: "test-icon",
    date: "12-12-2012",
  },
  native: false,
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <LikesItem { ...newProps } />;
};

beforeEach(() => {
  reset();
  startBuffering();
});

afterEach(() => {
  reset();
});

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("imageclasses returns array of css classes", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().imageclasses).toMatchSnapshot();
});

it("containerClasses returns non native version", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().containerClasses()).toMatchSnapshot();
});

it("containerClasses returns native version", () => {
  const wrapper = shallow(generateComponent({
    native: true,
  }));
  expect(wrapper.instance().containerClasses()).toMatchSnapshot();
});

it("preloader renders", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().preloader()).toMatchSnapshot();
});

it("renderElement renders", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().renderElement()).toMatchSnapshot();
});

it("onClick calls inAppLink with event", () => {
  const mockEvent = {
    currentTarget: {
      href: "https://test.com",
    },
  };
  const wrapper = shallow(generateComponent());
  wrapper.instance().onClick(mockEvent);
  expect(inAppLink).toHaveBeenCalledTimes(1);
  expect(inAppLink).toHaveBeenCalledWith(mockEvent);
  inAppLink.mockClear();
});

it("onClick does not call inAppLink if localhost link", () => {
  const mockEvent = {
    currentTarget: {
      href: "https://localhost.com",
    },
  };
  const wrapper = shallow(generateComponent());
  wrapper.instance().onClick(mockEvent);
  expect(inAppLink).not.toHaveBeenCalled();
  inAppLink.mockClear();
});

it("getDate returns formatted date", () => {
  const mockEntry = {
    date: "12-12-2012",
  };
  const wrapper = shallow(generateComponent());
  const result = wrapper.instance().getDate(mockEntry);
  expect(result).toBe("Dec 12, 2012");
});

it("getDate returns shortened date if this year", () => {
  const mockEntry = {
    date: "12-12-2000",
  };
  const wrapper = shallow(generateComponent());
  const result = wrapper.instance().getDate(mockEntry);
  expect(result).toBe("Dec 12");
});

it("iconClasses returns css class string", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().iconClasses).toMatchSnapshot();
});
