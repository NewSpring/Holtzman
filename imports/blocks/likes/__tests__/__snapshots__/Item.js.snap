exports[`test containerClasses returns native version 1`] = `"grid__item one-whole one-half@palm-wide-and-up"`;

exports[`test containerClasses returns non native version 1`] = `"grid__item one-whole"`;

exports[`test iconClasses returns css class string 1`] = `"test-icon soft-half-right"`;

exports[`test imageclasses returns array of css classes 1`] = `
Array [
  "background--fill",
  "card__image",
  "ratio--landscape",
]
`;

exports[`test preloader renders 1`] = `
<div
  className="background--fill card__image ratio--landscape load-item_1vqkxyp" />
`;

exports[`test renderElement renders 1`] = `
<div
  className="background--fill card__image ratio--landscape"
  style={
    Object {
      "backgroundImage": "url(\'http://test.com/1.jpg\')",
    }
  } />
`;

exports[`test renders with props 1`] = `
<div
  className="grid__item one-whole">
  <div
    className="card">
    <Link
      className="plain"
      onClick={[Function]}
      onlyActiveOnIndex={false}
      style={Object {}}
      to="http://test.com/1">
      <ImageLoader
        preloader={[Function]}
        renderElement={[Function]}
        src="http://test.com/1.jpg"
        wrapper={[Function]} />
      <div
        className="card__item soft text-dark-tertiary">
        <h4
          className="text-dark-primary capitalize" />
        <i
          className="test-icon soft-half-right" />
        <h7>
          test category
        </h7>
        <h7
          className="text-right float-right">
          Dec 12, 2012
        </h7>
      </div>
    </Link>
  </div>
</div>
`;
